From 757c96bf2b1ec491b399c1b7249158b2e612da61 Mon Sep 17 00:00:00 2001
From: "Li, Yifan" <yifan2.li@intel.com>
Date: Tue, 16 Jun 2020 18:10:45 +0800
Subject: [PATCH 1194/1214] Revert "ASoC: Intel: Skylake: Fix for multiple
 custom module formats"

This reverts commit a3a1e5323aeac97cec2eb2eca7669642ba7a0a17.
Due to CoE request on PKT-3528, revert this commit.

Tracked-On: PKT-3447
---
 sound/soc/intel/skylake/skl-messages.c | 18 +++++++++---------
 sound/soc/intel/skylake/skl-topology.c |  4 ++--
 sound/soc/intel/skylake/skl-topology.h |  2 ++
 3 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index 619e6f4..53a8dc0 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -1473,11 +1473,11 @@ static void skl_set_base_ext_module_format(struct skl_sst *ctx,
 	struct skl_module_iface *fmt = &module->formats[mconfig->fmt_idx];
 	struct skl_module_fmt *format;
 
-	base_cfg_ext->nr_input_pins = fmt->nr_in_fmt;
-	base_cfg_ext->nr_output_pins = fmt->nr_out_fmt;
+	base_cfg_ext->nr_input_pins = res->nr_input_pins;
+	base_cfg_ext->nr_output_pins = res->nr_output_pins;
 	base_cfg_ext->priv_param_length = 0;
 
-	for (i = 0; i < fmt->nr_in_fmt; i++) {
+	for (i = 0; i < res->nr_input_pins; i++) {
 		pin_res = &res->input[i];
 		pin_fmt = &base_cfg_ext->pins_fmt[i];
 
@@ -1488,9 +1488,9 @@ static void skl_set_base_ext_module_format(struct skl_sst *ctx,
 		fill_pin_params(&pin_fmt->audio_fmt, format);
 	}
 
-	for (i = 0; i < fmt->nr_out_fmt; i++) {
+	for (i = 0; i < res->nr_output_pins; i++) {
 		pin_res = &res->output[i];
-		pin_fmt = &base_cfg_ext->pins_fmt[fmt->nr_in_fmt + i];
+		pin_fmt = &base_cfg_ext->pins_fmt[res->nr_input_pins + i];
 
 		pin_fmt->pin_idx = pin_res->pin_index;
 		pin_fmt->buf_size = pin_res->buf_size;
@@ -1989,8 +1989,8 @@ static u16 skl_get_module_param_size(struct skl_sst *ctx,
 
 	case SKL_MODULE_TYPE_ALGO:
 		param_size = sizeof(struct skl_algo_cfg) +
-			     (m_intf->nr_in_fmt + m_intf->nr_out_fmt) *
-				     sizeof(struct skl_pin_format);
+			(m_res->nr_input_pins + m_res->nr_output_pins)
+			* sizeof(struct skl_pin_format);
 		param_size += mconfig->formats_config[SKL_PARAM_INIT].caps_size;
 		return param_size;
 
@@ -2008,8 +2008,8 @@ static u16 skl_get_module_param_size(struct skl_sst *ctx,
 	case SKL_MODULE_TYPE_BASE_GENEXT:
 		param_size = sizeof(struct skl_base_cfg);
 		param_size += sizeof(struct skl_base_cfg_ext) +
-			      (m_intf->nr_in_fmt + m_intf->nr_out_fmt) *
-				      sizeof(struct skl_pin_format);
+			(m_res->nr_input_pins + m_res->nr_output_pins)
+			* sizeof(struct skl_pin_format);
 		return param_size;
 
 	default:
diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index 0f55460..805c7c1 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -4934,14 +4934,14 @@ static int skl_tplg_get_int_tkn(struct device *dev,
 		if (!fmt)
 			return -EINVAL;
 
-		fmt->nr_in_fmt = tkn_elem->value;
+		res->nr_input_pins = tkn_elem->value;
 		break;
 
 	case SKL_TKN_MM_U32_NUM_OUT_FMT:
 		if (!fmt)
 			return -EINVAL;
 
-		fmt->nr_out_fmt = tkn_elem->value;
+		res->nr_output_pins = tkn_elem->value;
 		break;
 
 	case SKL_TKN_U32_FMT_CH:
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index 3784f2c..55aed1a 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -409,6 +409,8 @@ struct skl_module_res {
 	u32 obs;
 	u32 dma_buffer_size;
 	u32 cpc;
+	u8 nr_input_pins;
+	u8 nr_output_pins;
 	struct skl_module_pin_resources input[MAX_IN_QUEUE];
 	struct skl_module_pin_resources output[MAX_OUT_QUEUE];
 };
-- 
2.7.4

